test/operators/test_spin_op.py

```diff
@@ -0,0 +1,137 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test for SpinOp"""
+
+import unittest
+from itertools import product
+from test import QiskitNatureTestCase
+
+import numpy as np
+from ddt import data, ddt
+
+from qiskit.quantum_info import Pauli
+from qiskit_nature.operators import SpinOp
+
+
+def spin_labels(length):
+    """Generate list of fermion labels with given length."""
+    return ["".join(label) for label in product(["I", "X", "Y", "Z"], repeat=length)]
```

Comment:
Please also test `+` and `-`

Reply:
The test is [here](https://github.com/Qiskit/qiskit-nature/pull/35/files/564d221c2c4a288ba967673aaccce54369a1228b#diff-e1fad0f243ef03387da83f3f9c9b006212b413a01addd08569d3d78048469dd1R65-R80) since the logic of test is different.

Reply:
Alright, but please also add a variant where you test `+` and `-` in the dense format.

Reply:
OK, thanks